src/main/java/com/google/sps/utilities/RealTutorSessionDatastore.java

```diff
@@ -232,4 +257,32 @@ private long updateTimeRange(String email, TimeRange time, DatastoreService data
         return timeEntity.getKey().getId();
     }
 
+    /**
+    * Updates the TimeRange entity that the student canceled to reflect that it is now available and not scheduled.
+    * @return long, the id of the TimeRange entity
+    */
+    private long updateTimeRangeToAvailable(String email, TimeRange time, DatastoreService datastore, Transaction txn) {
+        //filter by tutor's email and time range properties
+        CompositeFilter timeFilter = CompositeFilterOperator.and(FilterOperator.EQUAL.of("email", "scheduled"),
```

Comment:
I didn't think of this before, but this might cause some issues because there could be multiple scheduled tutoring sessions at the same time. Do you think there's another way of doing this?

I'll be replacing the email field in the TimeRange entity with a tutorID field, so maybe instead of the tutor id, we can put the tutor session id for scheduled time ranges and filter using that?